code-search/code-search.ipynb

```diff
@@ -0,0 +1,609 @@
+        "    name = inflection.humanize(inflection.underscore(chunk[\"name\"]))\n",
+        "    signature = inflection.humanize(inflection.underscore(chunk[\"signature\"]))\n",
+        "\n",
+        "    # Check if docstring is provided\n",
+        "    docstring = \"\"\n",
+        "    if chunk[\"docstring\"]:\n",
+        "        docstring = f\"that does {chunk['docstring']} \"\n",
+        "\n",
+        "    # Extract the location of that snippet of code\n",
+        "    context = (\n",
+        "        f\"module {chunk['context']['module']} \"\n",
+        "        f\"file {chunk['context']['file_name']}\"\n",
+        "    )\n",
+        "    if chunk[\"context\"][\"struct_name\"]:\n",
+        "        struct_name = inflection.humanize(\n",
+        "            inflection.underscore(chunk[\"context\"][\"struct_name\"])\n",
+        "        )\n",
+        "        context = f\"defined in struct {struct_name} {context}\"\n",
+        "\n",
+        "    # Combine all the bits and pieces together\n",
+        "    text_representation = (\n",
+        "        f\"{chunk['code_type']} {name} \"\n",
+        "        f\"{docstring}\"\n",
+        "        f\"defined as {signature} \"\n",
+        "        f\"{context}\"\n",
+        "    )\n",
+        "\n",
+        "    # Remove any special characters and concatenate the tokens\n",
+        "    tokens = re.split(r\"\\W\", text_representation)\n",
+        "    tokens = filter(lambda x: x, tokens)\n",
+        "    return \" \".join(tokens)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Here is how the same structure looks like, after performing the normalization step:"
+      ],
+      "metadata": {
+        "id": "CDyJEb8WTHRP"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "textify(structures[0])"
+      ],
+      "metadata": {
+        "id": "dpkbii8gRvJW"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Let's do it for all the structures at once:"
+      ],
+      "metadata": {
+        "id": "_CYSPs_YTMyI"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "9tIgP_omkBYT"
+      },
+      "outputs": [],
+      "source": [
+        "text_representations = list(map(textify, structures))"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Created text representations might be directly used as an input to the `all-MiniLM-L6-v2` model."
+      ],
+      "metadata": {
+        "id": "O4oN7qOjTWgq"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "2DFkoAUSiW0Z"
+      },
+      "outputs": [],
+      "source": [
+        "from sentence_transformers import SentenceTransformer\n",
```

Comment:
Would really appreciate it if we used FastEmbed here — FastEmbed has this sentence transformer model as well, and it's a tad bit faster too!

Reply:
I initially used FastEmbed, but I wanted unified access to both models, and Jina code embeddings are not supported, right?